test/intl402/NumberFormat/prototype/formatRangeToParts/en-US.js

```diff
@@ -0,0 +1,45 @@
+// Copyright (C) 2021 Igalia, S.L. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-Intl.NumberFormat-formatRangeToParts
+description: Basic tests for the en-US output of formatRangeToParts()
+locale: [en-US]
+features: [Intl.NumberFormat-formatRangeToParts]
+---*/
+
+// Utils functions
+function* zip(a, b) {
+  assert.sameValue(a.length, b.length);
+  for (let i = 0; i < a.length; ++i) {
+    yield [i, a[i], b[i]];
+  }
+}
+
+function compare(actual, expected) {
+  for (const [i, actualEntry, expectedEntry] of zip(actual, expected)) {
+    assert.sameValue(actualEntry.type, expectedEntry.type, `type for entry ${i}`);
+    assert.sameValue(actualEntry.value, expectedEntry.value, `value for entry ${i}`);
+    assert.sameValue(actualEntry.source, expectedEntry.source, `source for entry ${i}`);
+  }
+}
+
+const nf = new Intl.NumberFormat("en-US", {signDisplay: "exceptZero"});
+
+// Perform ? RequireInternalSlot(nf, [[InitializedNumberFormat]]).
+let f = nf['formatRangeToParts'];
+assert.throws(TypeError, () => { f(1, 23) });
+
+// Basic example test en-US
+const nf2 = new Intl.NumberFormat("en-US", {
+  style: "currency",
+  currency: "EUR",
+  maximumFractionDigits: 0,
+});
+
+compare(nf2.formatRangeToParts(3, 5), [
+  {type: "currency", value: "€", source: "startRange"},
+  {type: "integer", value: "3", source: "startRange"},
+  {type: "literal", value: "–", source: "shared"},
+  {type: "integer", value: "5", source: "endRange"},
```

Comment:
Note that this is a different string than the one you have in the `formatRange` test.  I think ICU does this version, so you should fix the `formatRange` version.

Reply:
Good catch, thanks and done

Reply:
@sffc trying on FF
![image](https://user-images.githubusercontent.com/1572026/143567129-6c27e163-6293-4e85-9b44-1d4fa37cbc2f.png)
and 

![image](https://user-images.githubusercontent.com/1572026/143567192-8fd85ace-fba1-4c1e-80d0-6537852f296c.png)

Reply:
```js
const nf = new Intl.NumberFormat("en-US", {
  style: "currency",
  currency: "USD",
  maximumFractionDigits: 0,
});

console.log(nf.formatRange(3, 5)); // → "€3 – €5"
console.log(nf.formatRange(2.9, 3.1)); // → "~€3"
```

```js
const nf = new Intl.NumberFormat("es-ES", {
  style: "currency",
  currency: "EUR",
  maximumFractionDigits: 0,
});

console.log(nf.formatRange(3, 5); // → "3-5 €"
console.log(nf.formatRange(2.9, 3.1); // → "~3 €"
```